### PR TITLE
chore: determined_master_host and friends helm support, better defaults

### PR DESCRIPTION
**Pull request by stoksc**
## Ticket
DFR-530 / DFR-529

## Description
When `determined_master_ip` is unsettable via Helm and defaults to the service IP, life with proxies is hard. This change renames `determined_master_ip` to `determined_master_host` everywhere with some backwards compatibility, defaults `determined_master_host` to `<det_namespace>.<det_service_name>.svc.cluster.local`, and makes all of this overridable in Helm.

## Test Plan
Tested manually.

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/config/resource_manager_config.go [hunk 1]
```diff
@@ -188,8 +188,11 @@ type KubernetesResourceManagerConfig struct {
 	KubeconfigPath string       `json:"kubeconfig_path"`
 
 	DetMasterScheme string `json:"determined_master_scheme,omitempty"`
-	DetMasterIP     string `json:"determined_master_ip,omitempty"`
-	DetMasterPort   int32  `json:"determined_master_port,omitempty"`
+	// DeprecatedDetMasterHost shouldn't be used. Use the method DetMasterHost instead.
+	DeprecatedDetMasterHost string `json:"determined_master_host,omitempty"`
```
Comment (stoksc) [inline master/internal/config/resource_manager_config.go:192]: @maxrussell if you know a better trick for this let me know

File: master/cmd/determined-master/root.go [hunk 2]
```diff
@@ -321,6 +315,55 @@ func applyBackwardsCompatibility(configMap map[string]interface{}) (map[string]i
...
+		if rm["type"] != "kubernetes" {
+			return
+		}
+		if ip, ok := rm["determined_master_ip"]; ok {
+			if _, ok := rm["determined_master_host"]; !ok {
+				rm["determined_master_host"] = ip
+				delete(rm, "determined_master_ip")
+			} else {
+				log.Warn("ignoring duplicated configuration `determined_master_ip`")
+				delete(rm, "determined_master_ip")
```
Comment (maxrussell) [inline master/cmd/determined-master/root.go:352]: Hah, don't see a lot of `delete` in Go.

File: master/cmd/determined-master/root_test.go [hunk 3]
```diff
@@ -228,6 +228,60 @@ func TestApplyBackwardsCompatibility(t *testing.T) {
...
+					"determined_master_host": "10.0.0.1",
+				},
+				"additional_resource_managers": []map[string]any{
+					{
+						"type":                   "kubernetes",
+						"determined_master_host": "10.0.0.2",
+					},
+				},
+			},
+		},
```
Comment (maxrussell) [inline master/cmd/determined-master/root_test.go:284]: Maybe a test that ensures `determined_master_host` is preferred when they're both set?

**Review by maxrussell — APPROVED** [review]
LGTM. Could add that suggested test case, but I think it's okay without.

File: master/cmd/determined-master/root_test.go (hunk 3 above)
Comment (stoksc) [reply master/cmd/determined-master/root_test.go:284]: super easy, i'll add it

**Pull request merged**